src/mono/wasm/runtime/strings.ts

```diff
@@ -77,7 +77,11 @@ export class StringDecoder {
             // When threading is enabled, TextDecoder does not accept a view of a
             // SharedArrayBuffer, we must make a copy of the array first.
             // See https://github.com/whatwg/encoding/issues/172
-            const subArray = typeof SharedArrayBuffer !== "undefined" && Module.HEAPU8.buffer instanceof SharedArrayBuffer
+        
+            // BEWARE: In some cases, `instanceof SharedArrayBuffer` returns false even though buffer is an SAB.
+            // Patch adapted from https://github.com/emscripten-core/emscripten/pull/16994
+            // See also https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toStringTag
+            const subArray = typeof SharedArrayBuffer !== "undefined" && Module.HEAPU8.buffer[Symbol.toStringTag] === "SharedArrayBuffer"
```

Comment:
toStringTag is the perfect solution for this I think 👍

Reply:
I think so too! Very clean and just as easy to write.